af/fastpath/fastpath/core.py

```diff
@@ -0,0 +1,574 @@
+#!/usr/bin/env python3
+# # -*- coding: utf-8 -*-
+
+
+"""
+OONI Fastpath
+
+See README.adoc
+
+"""
+
+# Compatible with Python3.6 and 3.7 - linted with Black
+# debdeps: python3-setuptools
+
+from argparse import ArgumentParser, Namespace
+from datetime import date, datetime, timedelta
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+import hashlib
+import logging
+import multiprocessing as mp
+import os
+import sys
+import time
+
+from systemd.journal import JournalHandler  # debdeps: python3-systemd
+
+import ujson  # debdeps: python3-ujson
+import lz4.frame as lz4frame  # debdeps: python3-lz4
+
+# Feeds measurements from Collectors over SSH
+import fastpath.sshfeeder as sshfeeder
+
+# Feeds measurements from S3
+import fastpath.s3feeder as s3feeder
+
+# Push measurements into Postgres
+import fastpath.db as db
+
+from fastpath.metrics import setup_metrics
+
+import fastpath.utils
+
+LOCALITY_VALS = ("general", "global", "country", "isp", "local")
+
+NUM_WORKERS = 15
+
+log = logging.getLogger("fastpath")
+metrics = setup_metrics(name="fastpath")
+
+conf = Namespace()
+fingerprints = None
+
+
+def parse_date(d):
+    return datetime.strptime(d, "%Y-%m-%d").date()
+
+
+def setup():
+    os.environ["TZ"] = "UTC"
+    global conf
+    ap = ArgumentParser(__doc__)
+    ap.add_argument("--start-day", type=lambda d: parse_date(d))
+    ap.add_argument("--end-day", type=lambda d: parse_date(d))
+    ap.add_argument("--devel", action="store_true", help="Devel mode")
+    ap.add_argument(
+        "--update",
+        action="store_true",
+        help="Update summaries and files instead of logging an error",
+    )
+    ap.add_argument("--interact", action="store_true", help="Interactive mode")
+    ap.add_argument(
+        "--stop-after", type=int, help="Stop after feeding N measurements", default=None
+    )
+    conf = ap.parse_args()
+    if conf.devel:
+        root = Path(os.getcwd())
+        format = "%(relativeCreated)d %(process)d %(levelname)s %(name)s %(message)s"
+        logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, format=format)
+    else:
+        root = Path("/")
+        log.addHandler(JournalHandler(SYSLOG_IDENTIFIER="fastpath"))
+        log.setLevel(logging.DEBUG)
+
+    conf.conffile = root / "etc/fastpath.conf"
+    log.info("Using conf file %r", conf.conffile)
+    conf.vardir = root / "var/lib/fastpath"
+    conf.cachedir = conf.vardir / "cache"
+    conf.s3cachedir = conf.cachedir / "s3"
+    conf.dfdir = conf.vardir / "dataframes"
+    conf.outdir = conf.vardir / "output"
+    conf.msmtdir = conf.outdir / "measurements"
```

Comment:
`pathlib.Path` is pretty neat,  I didn't know of this.

Reply:
It even has convenient functions like Path(...).write_text(...)